backend/Controller.js

```diff
@@ -28,3 +31,64 @@ exports.getUser = async (req, res) => {
         })
 }
 
+//middleware for user login
+//handles both new and returning user login
+exports.loginUser = async (req, res) => {
+    //new user object to be populated with spotify api response data
+    let newUser = {}
+
+    //set refresh token
+    spotifyApi.setRefreshToken(req.body.refreshToken);
+
+    //set new access token
+    spotifyApi.refreshAccessToken().then(
+        (data) => {
+            spotifyApi.setAccessToken(data.body.access_token);
+
+            //get user object from SpotifyAPI
+            spotifyApi.getMe().then(
+                (data) => {
+                    //populate newUser object
+                    newUser.userID = data.body.id;
+                    newUser.refreshToken = req.body.refreshToken;
+                    newUser.name = data.body.display_name;
+                    newUser.imageURL = data.body.images[0].url;
+                    newUser.email = data.body.email;
+
+                    //delete previous mongodb entry
+                    //define query to find user with ID that matches ID for current request
+                    User.findOneAndDelete({ "userID": data.body.id }, { "sort": { "userID": 1 } })
+                        .then(() => {
+                            console.log("ID of login attempt: " + data.body.id);
+
+                            //create new User in MongoDB
+                            User.create(newUser).then(
+                                data => {
+                                    res.json(data);
+                                },
+                                err => {
+                                    res.json(err);
+                                }
+                            );
+                        },
+                            //error in user delete
+                            err => {
+                                res.json(err);
+                            }
+                        );
+                },
+                //SpotifyAPI return error
+                (err) => {
+                    res.status(400).json(err);
+                })
+        },
+        //access token refresh error
+        (err) => {
+            res.json("Unable to refresh access token.");
+            console.log(err);
+        }
+    );
+
+
```

Comment:
extra empty line (delete for formatting??)

Reply:
Fixed!